File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();
        
        for (int num : nums1) {
            set1.add(num);
        }
        
        for (int num : nums2) {
            set2.add(num);
        }
        
        List<Integer> ans1 = new ArrayList<>();
        List<Integer> ans2 = new ArrayList<>();
        
        for (int num : set1) {
            if (!set2.contains(num)) {
                ans1.add(num);
            }
        }
        
        for (int num : set2) {
            if (!set1.contains(num)) {
                ans2.add(num);
            }
        }
        
        List<List<Integer>> result = new ArrayList<>();
        result.add(ans1);
        result.add(ans2);
        
        return result;
    }
}

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)
        
        ans1 = [num for num in set1 if num not in set2]
        ans2 = [num for num in set2 if num not in set1]
        
        return [ans1, ans2]